tests/testsuite/metadata.rs

```diff
@@ -86,6 +86,185 @@ fn cargo_metadata_simple() {
         .run();
 }
 
+#[cargo_test]
+fn cargo_metadata_with_registered_dep() {
```

Comment:
I am not sure if this test case is necessary. But we don't have a test for feature names. So I added it. Please let me know if we have a better way to cover it.

Reply:
I dropped it. It is not a test for the real case, so it might not be worth adding it.